api-scripts/check-and-create-vocabulary-tags.py

```diff
@@ -0,0 +1,185 @@
+"""A script that checks and creates CKAN vocabulary tags.
+
+When run with ``create=False`` (default behavior), the script will check
+the existing tags in the Vocabulary associated with the provided gmm yml key
+and return a list of tags in the yml that don't currently exist in the
+Vocabulary.
+
+When run with ``create=True``, any tags that do not currently exist in the
+Vocabulary will be created. Tags are standardized to all-uppercase and
+(per CKAN requirements) must be strings between 2 and 100 characters long,
+containing only alphanumeric characters and hyphens, underscores, and periods.
+
+Dependencies:
+    $ mamaba install ckanapi pyyaml requests
+"""
+import argparse
+import logging
+import os
+
+import ckanapi.errors
+import requests  # mamba install requests
+import yaml  # mamba install pyyaml
+from ckanapi import RemoteCKAN  # mamba install ckanapi
+
+
+logging.basicConfig(level=logging.DEBUG)
+LOGGER = logging.getLogger(os.path.basename(__file__))
+CKAN_HOSTS = {
+    'prod': 'https://data.naturalcapitalproject.stanford.edu',
+    'staging': 'https://data-staging.naturalcapitalproject.org',
+    'dev': 'https://localhost:8443'
+}
+CKAN_APIKEY_ENVVARS = {
+    'prod': 'CKAN_APIKEY',
+    'staging': 'CKAN_STAGING_APIKEY',
+    'dev': 'CKAN_LOCAL_APIKEY',
+}
+assert set(CKAN_HOSTS.keys()) == set(CKAN_APIKEY_ENVVARS.keys()), (
+    'Mismatch between keys in CKAN host and apikey dicts')
+
+
+# Map of GMM YAML KEYS to the associated CKAN Tag Vocabulary names
+YML_KEY_VOCAB_NAME_MAP = {
+    'placenames': 'place'
+}
+
+
+def main(ckan_url, ckan_apikey, gmm_yaml_path, yaml_key, create=False,
+         verify_ssl=True):
```

Comment:
Would it be possible to add a docstring here, mostly to describe the input parameters' purpose and type, for future us?